freqtrade/exchange/exchange.py

```diff
@@ -2283,6 +2303,146 @@ async def _fetch_funding_rate_history(
+
+                        if not all_stored_ticks_df.empty:
+                            if all_stored_ticks_df.iloc[0]['timestamp'] <= first_candle_ms:
+                                last_cached_ms = all_stored_ticks_df.iloc[-1]['timestamp']
+                                # only use cached if it's closer than first_candle_ms
+                                since_ms = last_cached_ms if last_cached_ms > first_candle_ms else first_candle_ms  # noqa
+                            # doesn't go far enough
+                            else:
+                                all_stored_ticks_df = DataFrame(
+                                    columns=DEFAULT_TRADES_COLUMNS + ['date'])
+
+                    # from_id overrules with exchange set to id paginate
+                    # TODO: DEBUG:
+                    # since_ms = 1698060269000
+                    # from_id = None
+                    # TODO: /DEBUG
+                    [ticks_pair, new_ticks] = self.get_historic_trades(pair,
+                                                                       since=since_ms if since_ms else first_candle_ms, # noqa
+                                                                       until=until,
+                                                                       from_id=from_id)
+
+                except Exception as e:
+                    logger.error(f"Refreshing TRADES data for {pair} failed")
+                    logger.error(e)
+                    raise e
+
+                if new_ticks:
+                    all_stored_ticks_list = all_stored_ticks_df[DEFAULT_TRADES_COLUMNS].values.tolist()  # noqa: E501
+                    all_stored_ticks_list.extend(new_ticks)
+                    trades_df = self._process_trades_df(pair,
+                                                        timeframe,
+                                                        candle_type,
+                                                        all_stored_ticks_list,
+                                                        cache,
+                                                        first_required_candle_date=first_candle_ms)
+                    results_df[(pair, timeframe, candle_type)] = trades_df
+                    data_handler.trades_store(f"{pair}-cached", trades_df[DEFAULT_TRADES_COLUMNS])
+
+                else:
+                    raise OperationalException("no new ticks")
+
+        return results_df
+
+    def _now_is_time_to_refresh_trades(self,
+                                       pair: str,
+                                       timeframe: str,
+                                       candle_type: CandleType) -> bool:
+        # Timeframe in seconds
+        df = self.klines((pair, timeframe, candle_type), True)
+        _calculate_ohlcv_candle_start_and_end(df, timeframe)
+        timeframe_to_seconds(timeframe)
+        plr = round(df.iloc[-1]["candle_end"].timestamp())
+        now = int(timeframe_to_prev_date(timeframe).timestamp())
```

Comment:
I think if you reverse this logic - it can be simplified quite significantly (including the removal of the "copy" argument on `klines()`).

simply getting the last candle date (`df.iloc[-1]['date']`) - and then adding 1x timeframe to it will suffice. 
`_calculate_ohlcv_candle_start_and_end()` may be useful - but it's doing the calculation on 1000 candles - while we only need it on the most recent one.